src/components/tvShow/tvShow.js

```diff
@@ -11,10 +11,12 @@ class TVShow extends Component {
       fetchState: null,
       show: null,
     };
+
+    this.onClick = this.onClick.bind(this);
   }
 
-  componentDidMount() {
-    const { showId } = this.props.match.params;
+  componentWillMount() {
```

Comment:
I put all fetching data to componentWillMount, but not sure if it's a good place for it